tests/test_urls.py

```diff
@@ -119,15 +121,17 @@ def test_missing_parameters_for_new_issue(self):
 
     def test_new_issue_should_not_crash(self):
         '''Checks 500 is not accepted for /issues/new POST.'''
-        data = {'problem_category': u'mobile_site_bug',
-                'username': u'',
-                'description': u'foo',
-                'submit-type': u'github-proxy-report',
-                'url': u'http://example.com',
-                'os': u'Foobar',
-                'browser': u'BarFoo'}
-        rv = self.app.post('/issues/new', data=data)
-        self.assertNotEqual(rv.status_code, 500)
+        with patch('requests.post') as github_data:
+            github_data.status_code.return_value = 500
+            data = {'problem_category': u'mobile_site_bug',
+                    'username': u'',
+                    'description': u'foo',
+                    'submit-type': u'github-proxy-report',
+                    'url': u'http://example.com',
+                    'os': u'Foobar',
+                    'browser': u'BarFoo'}
+            rv = self.app.post('/issues/new', data=data)
+            self.assertNotEqual(rv.status_code, 500)
```

Comment:
That's weird, for me this is calling the GitHub API so much that it's even posting issues: https://github.com/webcompat/webcompat-tests/issues/1239 . 

I just understood that it shouldn't be posting anything, in fact the posting of the issue should fail since we are sending a "bad request".

Just to see if I got it right: this test's response should return a 400 status code, right? Because the request should abort before going to GitHub.

If I'm right, the test is giving me a false positive and we could change ` self.assertNotEqual(rv.status_code, 500)` to ` self.assertEqual(rv.status_code, 400)` just so that we avoid that.

I just don't understand why it's posting the issue with my user when the request has the username field empty. I'll look into that and edit this with my findings later.